target/newer.go

```diff
@@ -0,0 +1,126 @@
+package target
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	// errNewer is an ugly sentinel error to cause filepath.Walk to abort
+	// as soon as a newer file is encountered
+	errNewer = fmt.Errorf("newer item encountered")
+)
+
+// DirNewer reports whether any item in sources is newer than the target time.
+// Sources are searched recursively and searching stops as soon as any entry
+// is newer than the target.
+func DirNewer(target time.Time, sources ...string) (bool, error) {
+	walkFn := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.ModTime().After(target) {
+			return errNewer
+		}
+		return nil
+	}
+	for _, source := range sources {
+		source = os.ExpandEnv(source)
+		err := filepath.Walk(source, walkFn)
+		if err == nil {
+			continue
+		}
+		if errors.Is(err, errNewer) {
```

Comment:
So I don't think filepath munges the error, and errors.Is isn't in older versions of Go, so I think I'd rather if this were just a plain old comparison.

```suggestion
		if err == errNewer {
```

Reply:
done